controllers/catalogentry_controller.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	catalogv1alpha1 "github.com/kcp-dev/catalog/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// CatalogEntryReconciler reconciles a CatalogEntry object
+type CatalogEntryReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=catalog.kcp.dev,resources=catalogentries,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Does the catalog entry controller need non-finalizer write permissions on catalog entry objects?

Reply:
I don't see why not. The controller will need to write into the status.

Reply:
You have that already with `catalogentries/status` subresource update/patch permission. I think you can drop `create/update/patch/delete` on the main resource RBAC unless there's something about the controller I'm not aware of.

Reply:
I'm dropping create and delete but will keep update/patch in case we need to update certain labels and annotations.